js/main.js

```diff
@@ -18,16 +19,42 @@ function submitForm(event) {
     notes: formElements.elements.notes.value,
     entryId: data.nextEntryId,
   };
-  data.nextEntryId++;
-  data.entries.unshift(formEntries);
-  imgElement.src = '../images/placeholder-image-square.jpg';
-  form.reset();
-  const unorderedList = document.querySelector('ul');
-  unorderedList.appendChild(renderEntry(formEntries));
-  viewSwap('entries');
-  if (localStorage.getItem('form-submission') === null) {
-    toggleNoEntries();
+
+  if (data.editing === null) {
+    data.nextEntryId++;
+    data.entries.unshift(formEntries);
+    imgElement.src = '../images/placeholder-image-square.jpg';
+    form.reset();
+    const unorderedList = document.querySelector('ul');
+    unorderedList.appendChild(renderEntry(formEntries));
+    viewSwap('entries');
+    if (localStorage.getItem('form-submission') === null) {
+      toggleNoEntries();
+    }
   }
+  if (data.editing !== null) {
+    formEntries.entryId = data.editing.entryId;
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === data.editing.entryId) {
+        data.entries[i] = formEntries;
+      }
+    }
+
+    const currentLiElements = document.querySelectorAll('li');
+    for (let i = 0; i < currentLiElements.length; i++) {
+      if (
+        currentLiElements[i].getAttribute('data-entry-id') ==
+        data.editing.entryId
+      ) {
+        currentLiElements[i] = renderEntry(formEntries);
```

Comment:
use the `replaceWith` method here to get the list to update in real time without refreshing the page.